src/traces/box/cross_trace_calc.js

```diff
@@ -90,19 +95,46 @@ function setPositionOffset(traceType, gd, boxList, posAxis, pad) {
     var groupgap = fullLayout[traceType + 'groupgap'];
     var padfactor = (1 - gap) * (1 - groupgap) * dPos / fullLayout[numKey];
 
-    // autoscale the x axis - including space for points if they're off the side
-    // TODO: this will overdo it if the outermost boxes don't have
-    // their points as far out as the other boxes
-    var extremes = Axes.findExtremes(posAxis, boxdv.vals, {
-        vpadminus: dPos + pad[0] * padfactor,
-        vpadplus: dPos + pad[1] * padfactor
-    });
+    // Find maximum trace width
+    // we baseline this at dPos
+    var maxHalfWidth = dPos;
+    for(i = 0; i < boxList.length; i++) {
+        calcTrace = calcdata[boxList[i]];
+
+        if(calcTrace[0].trace.width) {
+            if(calcTrace[0].trace.width / 2 > maxHalfWidth) {
+                maxHalfWidth = calcTrace[0].trace.width / 2;
+            }
+        }
+    }
 
     for(i = 0; i < boxList.length; i++) {
         calcTrace = calcdata[boxList[i]];
-        // set the width of all boxes
-        calcTrace[0].t.dPos = dPos;
-        // link extremes to all boxes
+        // set the width of this box
+        // override dPos with trace.width if present
+        var thisDPos = calcTrace[0].t.dPos = (calcTrace[0].trace.width / 2) || dPos;
+        var positions = calcTrace.map(getPosition);
+        // autoscale the x axis - including space for points if they're off the side
+        // TODO: this will overdo it if the outermost boxes don't have
+        // their points as far out as the other boxes
+        var vpadminus = 0;
+        var vpadplus = 0;
+        if(calcTrace[0].trace.side === 'positive') {
+            vpadminus = 0;
+            vpadplus = thisDPos + pad[1] * padfactor;
+        } else if(calcTrace[0].trace.side === 'negative') {
+            vpadminus = thisDPos + pad[0] * padfactor;
+            vpadplus = 0;
+        } else {
+            // if side === 'both'
+            vpadminus = thisDPos + pad[0] * padfactor;
+            vpadplus = thisDPos + pad[1] * padfactor;
+        }
```

Comment:
🌴 
```js
var side = calcTrace[0].trace.side;
var vpadminus = (side === 'positive') ? 0 : (thisDPos + pad[0] * padfactor);
var vpadplus = (side === 'negative') ? 0 : (thisDPos + pad[1] * padfactor);
```

Reply:
I'm a bit worried actually that this will exacerbate the issue in the TODO above. For example what if you have a single-sided violin but the points get drawn on the other side? It looks to me like this algorithm will cut them off entirely. @etpinard thoughts?

Reply:
> I'm a bit worried actually that this will exacerbate the issue in the TODO above

I noticed another bug - if we choose to show the box, it gets cut in half if we choose `positive` or `negative` for `side`
![half violin](https://user-images.githubusercontent.com/10369095/47882618-7e972100-de00-11e8-9d2c-2f23b73cadfe.png)

DATA:
```javascript
var data = [{
    type: 'violin',
    x: [0, 5, 7, 8],
    points: 'all',
    side: 'positive',
    box: {
      visible: true
    },
    boxpoints: true,
    line: {
      color: 'black'
    },
    fillcolor: '#8dd3c7',
    opacity: 0.6,
    meanline: {
      visible: true
    },
    y0: 0.0
}];
```

I think we are going to have to correlate the currently `0` vpad with the span of the box and visible points.

The fact that the image tests passed tells me an example like ^ is not in mocks.  I think we should add a single sided violin baseline.

Reply:
> What does 🌴 mean? _Don't repeat yourself_?

yeah sorry, DRY - I guess that's just a plotly.js convention.

> I noticed another bug - if we choose to show the box, it gets cut in half if we choose `positive` or `negative` for `side`

That's intentional, eg for two back-to-back violins. https://github.com/plotly/plotly.js/blob/master/test/image/baselines/violin_side-by-side.png